go/apps/http_api_nostream/resource.py

```diff
@@ -247,6 +247,13 @@ def handle_PUT_in_reply_to(self, request, payload, in_reply_to):
     @inlineCallbacks
     def handle_PUT_send_to(self, request, payload):
         user_account = request.getUser()
+        optout_store = self.worker.vumi_api.get_user_api(user_account) \
+            .optout_store
+        optout = yield optout_store.get_opt_out(
+            payload.get('to_addr_type', 'msisdn'), payload.get('to_addr'))
+        if optout:
+            self.client_error_response(request, "Recipient has opted out")
```

Comment:
Wondering if this case should be counted as an error case. I would guess people using the api would just expect us to send to non-opted out users. It does seem like we should be saying something though.

If we are going to send some kind of response for the opt out cases, maybe we should be including the address of the opted out person and mention more clearly what happened, something like: "Ignoring opted-out recipient +2731234567".

Thoughts?